### PR TITLE
build-deploy documentation

### PR DESCRIPTION
**Pull request by nilsgstrabo**

### CODE REVIEW

**Review by emirgens — COMMENTED** [review]
Too much text for me to understand, but see my comment in text

File: public-site/docs/src/guides/build-and-deploy/index.md [hunk 1]
```diff
@@ -4,28 +4,97 @@ title: Build and deploy
...
+      dockerfileName: bar.Dockerfile
+      src: .
+```
+
+#### Summary
+
+- Pipelines created manually from `Radix Web Console` or `Radix CLI` will always build all components and jobs.
+- When pipelines are created from a `Github webhook`, Radix will only build components and jobs that affected by the changes, and reuse images from the active deployment for unchanged components/jobs.
+- When Radix detects that `radixconfig.yaml` is modified, all components and jobs are built.
+- When `build secret` values are updated, the next pipeline job will build all components and jobs.
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply public-site/docs/src/guides/build-and-deploy/index.md]: fixed

File: public-site/docs/src/guides/build-and-deploy/index.md [hunk 2]
```diff
@@ -4,28 +4,97 @@ title: Build and deploy
...
 
-There are few options how the code can be located in the GitHub repository for a Radix application:
+For manually created pipeline jobs, Radix will always build all components and jobs. When a pipeline job is created from a Github webhook, Radix compares the commit ID from the webhook request body with the commit ID of the active deployment to detect which directories have changed. The list of changed directories are then compared to the `src` property for each component and job, and if any of the changed directories are equal to, or a child of `src`, a new image is built for the component/job. Radix will reuse the image from the current active deployment for components and jobs that have not changed.
```
Comment (satr) [inline public-site/docs/src/guides/build-and-deploy/index.md:19]:
```suggestion
For manually created pipeline jobs, Radix will always build all components and jobs without defined property `image`. When a pipeline job is created by a Github webhook, Radix compares the commit ID from the webhook request body with the commit ID of the active deployment to detect which directories have changed. The list of changed directories are then compared to the `src` property for each component and job:
* If any of the changed directories are within  the directory, defined in `src` or are its sub-directories, a new image is built for the component and job. 
* if no such changes detected, Radix will reuse the image from the current active deployment for components and jobs.
```

File: public-site/docs/src/guides/build-and-deploy/index.md [hunk 3]
```diff
@@ -4,28 +4,105 @@ title: Build and deploy
...
+      - SECRET1
+      - SECRET2
+  components:
+    - name: foo
+      src: /foo
+    - name: bar
+      src: /bar
+```
+
+Components `foo` and `bar` are configured with different directories for their Docker files. We assume that an active deployment already exist, created from a `build-deploy` pipeline that built and deployed images named `foo:tag1` and `bar:tag1` for the `foo` and `bar` components respectively.
```
Comment (Richard87) [inline public-site/docs/src/guides/build-and-deploy/index.md:62]: remove space, "Docker files" => "Dockerfiles"?

File: public-site/docs/src/guides/build-and-deploy/index.md [hunk 4]
```diff
@@ -4,28 +4,123 @@ title: Build and deploy
...
+::: tip TL;DR
+- Pipeline jobs created manually from `Radix Web Console` or `Radix CLI` will always build all components and jobs.
+- For pipeline jobs created from a `Github webhook`, Radix will only build components and jobs that affected by the changes, and reuse images from the active deployment for unchanged components/jobs.
+- When Radix detects that `radixconfig.yaml` is modified, all components and jobs are built.
+- When values for `build secrets` defined in radixconfig.yaml, are updated, the next pipeline job will build all components and jobs.
+:::
 
-There are few options how the code can be located in the GitHub repository for a Radix application:
+The [`build-deploy`](../../start/workflows/) pipeline builds and deploys container images from Docker files for components and jobs that do not have the [`image`](../../references/reference-radix-config/#image) property set in [`radixconfig.yaml`](../../references/reference-radix-config). The location of the Docker file for each component and job is defined in the [`dockerfileName`](../../references/reference-radix-config/#dockerfilename) and [`src`](../../references/reference-radix-config/#src) properties.
```
Comment (Richard87) [inline public-site/docs/src/guides/build-and-deploy/index.md:15]: Change `Docker files` to `Dockerfiles`?

File: public-site/docs/src/guides/build-and-deploy/index.md (hunk 2 above)
Comment (nilsgstrabo) [reply public-site/docs/src/guides/build-and-deploy/index.md]: This is described in the previous paragraph. I don't think there is a need to describe it one more time.

**Review by Richard87 — CHANGES_REQUESTED** [review]
Looks good, tiny comment on word splitting (Dockerfiles)

File: public-site/docs/src/guides/build-and-deploy/index.md (hunk 4 above)
Comment (nilsgstrabo) [reply public-site/docs/src/guides/build-and-deploy/index.md]: fixed

**Review by Richard87 — DISMISSED** [review]
lgtm!

**Review by satr — DISMISSED** [review]

**Review by satr — APPROVED** [review]

**Pull request merged**